### PR TITLE
TB-3660: temporary fix for navBarItems overflow issue

### PR DESCRIPTION
**Pull request by pro100svitlo**
### What 🕵️ 🔍

- calculate if `NavBar` width is too big for the current view
- if so, apply reduced icon size

----------

### How to test 🥼 🔬

- switch to [this PR](https://github.com/xaynetwork/xayn_discovery_app/pull/455) and try it there
- you need to run it on the `iphone SE` simulator to validate result

----------

### Screenshots 📸 📱

| Before | After  |
| ------ | ------ |
|  |  |

----------

### References 📝 🔗

- [JIRA](https://xainag.atlassian.net/browse/TB-3660)

----------

### CODE REVIEW

File: lib/src/widget/nav_bar/widget/nav_bar.dart [hunk 1]
```diff
@@ -39,48 +39,93 @@ class NavBarState extends State<NavBar> implements ConfigUpdater {
...
+            navBarSidePadding * 2 +
+
+            // padding from left and right
+            widget.padding.left +
+            widget.padding.right;
+    while (shouldIconBeSmaller) {
+      iconSize -= 1;
+      _checkDesiredWidth(itemsCount);
+    }
+  }
```
Comment (joecks) [inline lib/src/widget/nav_bar/widget/nav_bar.dart:86]: 🛑 I am not sure you need to use a while loop:

```
maxPossibleWidth = (iconSize * itemCount) + paddings;
maxPossibleWidth - paddings = (iconSize * itemCount) 
(maxPossibleWidth - paddings) / itemCount = iconSize 

```

so 
`iconSize  = ((maxPossibleWidth - paddings) / itemCount).floor()`

Is a bit more cleaner and it avoids the dangerous while loop.

File: lib/src/widget/nav_bar/widget/nav_bar.dart [hunk 2]
```diff
@@ -39,48 +39,93 @@ class NavBarState extends State<NavBar> implements ConfigUpdater {
...
-        ? _withPadding(sized, MediaQuery.of(context).viewInsets.bottom)
-        : _withPadding(sized);
+  @override
+  Widget build(BuildContext context) {
+    final config = this.config;
+    if (config == null || config.type.isHidden) return const Center();
+
+    return LayoutBuilder(builder: (_, constrain) {
+      maxPossibleWidth = constrain.maxWidth;
+      _checkDesiredWidth(config.items.length);
```
Comment (joecks) [inline lib/src/widget/nav_bar/widget/nav_bar.dart:95]: ❗  To not hide the sideEffects I would propose to change this a bit

```
final newIconSize = _calculateNewIconSize(itemCount)
If (newIconSize < iconSize) iconSize = newIconSize
```

**Review by joecks — CHANGES_REQUESTED** [review]

**Comment by joecks** [comment]
So I tested it on a small screen like the iPhone 8. and it works as expected but I also observed a CRASH ... so I am not sure if something wasn't right with the while loop

File: lib/src/widget/nav_bar/widget/nav_bar.dart (hunk 1 above)
Comment (pro100svitlo) [reply lib/src/widget/nav_bar/widget/nav_bar.dart]: updated this code

File: lib/src/widget/nav_bar/widget/nav_bar.dart [hunk 3]
```diff
@@ -39,48 +39,93 @@ class NavBarState extends State<NavBar> implements ConfigUpdater {
...
-        boxShadow: [linden.styles.cardShadow],
-      ),
-    );
+  bool useDefaultIconSize = true;
+
+  bool get shouldIconBeSmaller {
+    final shouldBeSmaller = desiredWidth > maxPossibleWidth;
+    if (useDefaultIconSize && shouldBeSmaller) useDefaultIconSize = false;
+    return shouldBeSmaller;
+  }
```
[the remark this answers is not in the document]
Comment (pro100svitlo) [reply lib/src/widget/nav_bar/widget/nav_bar.dart]: this code was removed

**Review by joecks — APPROVED** [review]
There is some uncomented code that can be deleted

File: lib/src/widget/nav_bar/widget/nav_bar.dart [hunk 4]
```diff
@@ -39,48 +39,76 @@ class NavBarState extends State<NavBar> implements ConfigUpdater {
...
+
+  @override
+  Widget build(BuildContext context) {
+    final config = this.config;
+    if (config == null || config.type.isHidden) return const Center();
+
+    return LayoutBuilder(builder: (_, constrain) {
+      _updateCustomIconSize(constrain.maxWidth, config.items.length);
+      // _checkDesiredWidth(config.items.length);
```
[the remark this answers is not in the document]
Comment (pro100svitlo) [reply lib/src/widget/nav_bar/widget/nav_bar.dart]: removed

**Pull request merged**